### ISSUE TITLE
Filenames include query string making them invalid

### ISSUE THREAD

**Issue by jeppe-smith** [iFaxity/vite-plugin-istanbul#23]
I have an issue with the filenames containing an appended query string (eg. `src/views/UsersIndex.vue?vue&type=template&lang.js`) in the instrumentation. This causes our CI to fail when it tries to upload our coverage.

After looking around I changed transform method to the following to get rid of the query string.

```js
if (exclude.shouldInstrument(id)) {
  // Remove querys strings from file names.
  const filename = id.replace(/\?.+$/, '')
  const sourceMap = sanitizeSourceMap(this.getCombinedSourcemap());
  const code = instrumenter.instrumentSync(srcCode, filename, sourceMap);
  const map = instrumenter.lastSourceMap();

  // Required to cast to correct mapping value
  return { code, map } as TransformResult;
}
```

**Comment by iFaxity** [comment]
Hi @jeppe-smith, great finding!

I will create a pull request with this change and run some tests to see if it works in my setup aswell.

**Comment by iFaxity** [comment]
There appears to be an official way to get the file name from the reference id. As Vite uses Rollups plugin system in the background, it shares some of it's API: [this.getFilename](https://rollupjs.org/guide/en/#thisgetfilename). I will try this and see if it gets the desired results.

**Comment by iFaxity** [comment]
This will be fixed when the PR #27 gets merged. I'll hopefully do some testing during the weekend.
